### DISCUSSION TITLE
File extension of Optolith JSON export

### DISCUSSION THREAD

**Discussion by Nargaff** [elyukai/optolith-client#1249] [Ideas]
I'm the developer of the TDA character generator. Someone recently asked me if I could add JSON export to TDA to be able to use all the 3rd party tools that use the Optolith JSON export. The way the question was framed, I was not sure if the user understood that JSON is a generic format but has a very specific structure that depends on the application that creates it. Even if he got it right I think normal users do not necessarily know that JSON is not some 'standard' like DOCX. I could imagine that a lot of users would try to use JSON files that TDA creates in places where a Optolith JSON is actually needed. So I thought maybe it would be better if Optolith (and any other CG) would use a specific file extension instead of just .json. This could avoid confusion. Also it would spare 3rd party developers support calls from such users and the hassle to figure out why 'the JSON' file does not work. I guess there would also be downsides to dropping the JSON extension.

**Comment by pickbreaker** [comment]
IMHO if just a random extension was used it would be better to drop the file extension altogether. I could see it to lead to even more confusion if we used a random one.

**Reply by elyukai** [reply]
But how do you tell people then if a file is an Optolith character? They must remember they exported this exact file. Also, how do you describe in docs what an Optolith character file is?

**Reply by Zakant** [reply]
Especially for windows this is a really bad idea. The files will show up blank without any logo and only "file" as a description. A extension that is (on the machine) registered for optolith would show "optolith charakter file" as the description text which would be very clear.

**Comment by elyukai** [comment]
> Also ich habe durchaus schon versucht VTT-JSONs in Optolith zu laden und vice versa, aber habe das dann auch schnell bemerkt. Von dem finalen ".json" würde ich nicht weggehen, da es angenehmer ist, wenn man die Datei in einem Texteditor öffnet (dann wird direkt der präferierte Editor für JSON verwendet, ohne dass man das für Optolith-Dateien extra konfigurieren muss). Wenn du sowas vorhast, dann vielleicht ".optolith.json" oder ähnliches?

— @florian1345 on Discord

**Reply by elyukai** [reply]
I also thought of variants like `Alrik.character.json` or `Alrik.char.json`

**Comment by JoveToo** [comment]
I think a specific extension would be a plus. It would also make cross importing with different tools easier. Basically, that the internal format is json is irrelevant. You just want to know which tool to import it in.

**Comment by nitrout** [comment]
IMHO a special file extension would be a plus of usability for the average user.
The advantages of *.opto files (just an example extension) would be:
1. Users can easily recognize the character files and have a connection with Optolith
2. The structure of keys and values in the json data is known and 3rd party software can rely on that
3. 3rd party software can easily detect if a file valid because of the extension or just let users only select *.opto files for import
4. the internal processing of the files would not change (not really an advantage but not a disadvantage either ;) )
